### DISCUSSION TITLE
Gestione scuri

### DISCUSSION THREAD

**Discussion by Jocman** [indomus/forum#179] [HUB: Home Assistant]
Sto muovemdo i primi passi, ed ho optato per provare a domotizzare i miei scuri.
Preciso "scuri" e non "persiane/tapparelle" (che dir si voglia).
Situazione: 3 finestre con scuri da gestire; ogni finestra ha 2 pulsanti (1 apre e 1 chiude). Premendo uno dei 2 pulsanti lo scuro si apre/chiude completamente. (volendo, se si ripreme lo stesso pulsante durante la relativa operazione lo scuro si ferma alla posizione raggiunta)
Ho preso 3 Shelly 2.5, impostati come "shutter" con "pulsanti", e seguendo la guida Indomus li ho:
- dissociati dal cloud
-  abbinati a MQTT
- configurati come "cover" (beh, copia/incolla, facendo le dovute variazioni ai nomi....)
- impostato su lovelace la relativa scheda.
In effetti, i punti 4 e 5 li ho fatti solo per il primo Shelly (giusto per testare se funziona - non l'ho ancora neanche inserito fisicamente nella finestra)
Purtroppo non va.... Di seguito i punti che ritengo "critici"

A) Lo shelly l'ho impostato come "shutter". Ritengo che tale funzione sia studiata esclusivamente per gli avvolgibili, in quanto richiede di impostare il tempo di apertura / chiusura, e infatti dando il comando lo shelly "preme" il pulsante per il tempo necessario affinchè la tapparella salga/scenda completamente, in quanto una persona terrebbe premuto il relativo pulsante per tutto il tempo necessario.
I miei scuri invece richiedono una pressione istantanea del pulsante, per poi aprirsi/chiudersi in autonomia. Penso (penso) di aver bypassato questo inconveniente impostando il tempo di apertura/chiusura al minimo possibile (1 secondo), e infatti sento il rele attivarsi/disattivarsi in 1 secondo (questo dovrebbe simulare la mia pressione istantanea del pulsante a muro)

B) La scheda Lovelace "cover" anche è "avvolgibile oriented", riportando una freccia in su, uno stop e una freccia in giu. Poco male, dopotutto non mi crea grande problema ricordarmi che "su>apre - giu>chiude" lo scuro (lo stop in realtà mi è assolutamente inutile)
Purtroppo però ho notato che di 3 simboli, gli unici attivi sono lo STOP e GIU, mentre la freccia SU è disabilitata.

Dando (ottimisticamente) per scontato che il punto A sia la soluzione al problema "shutter mode", tuttavia mi resta il problema B, e cioè che il comando di apertura (SU) non è disponibile. Fosse stato lo STOP a non essere disponibile non era un problema (come detto non mi è necessario), ma lo scuro lo vorrei aprire.....

Andrea

**Comment by MarioBomb** [comment]
>  Penso (penso) di aver bypassato questo inconveniente impostando il tempo di apertura/chiusura al minimo possibile (1 secondo), e infatti sento il rele attivarsi/disattivarsi in 1 secondo (questo dovrebbe simulare la mia pressione istantanea del pulsante a muro)

grave errore, rifai la calibrazione, lascia i tempi invariati, anche le tapparelle salgono/scendono con pressione istantanea del tasto senza dover tenere premuto, basta configurarli nel modo corretto. Ma prima configura nell'interfaccia web dello shelly "BUTTON TYPE" toggle e in "INPUT buttons mode" = OPEN & CLOSE

**Reply by Jocman** [reply]
Innanzitutto grazie della risposta.
Ho variato i button type e input mode come mi hai suggerito (ho dovuto variare solo button type); e ho impostato i timer a 15 sec (tanto ci mettono gli scuri ad aprirsi/chiudersi).
Quando parli di "rifare la calibrazione" ti riferisci alla funzione "Positioning control > Enable > Calibrate"? Perchè ho provato a fare anche quella, ma ottengo  un errore "Minumum power 4w".....
Ma mi resta sempre il problema su Lovelace che il pulsante di apertura (SU) non è disponibile.
Questa è una immagine della scheda in questione:

![screenshot 6](https://user-images.githubusercontent.com/52747587/115265958-06d82080-a138-11eb-8537-de3d042e163a.jpg)

Nel primo post ho dimenticato di precisare che premendo sull'unica freccia disponibile (GIU),  mentre il rele rimane eccitato, la freccia SU si abilita (e GIU si diabilita), tornando alla situazione della foto quando il rele di diseccita.
Girovagando in rete, ho visto che anche altri  utenti hanno qualcosa di simile, spesso unito ad altri controlli tapparella che però mostrano tutti e 3 i pulsanti disponibili.

Potrei pensare che (più o meno giustamente) mi lascia attivata l'unica opzione valida al momento (se sono aperte, l'unica alternativa è chiuderle.....), ma allora mi aspetterei che una volta trascorso il tempo previsto (nel mio caso i 15 sec), mi disabilitasse GIU e lasciasse attivo SU (del resto sono chiuse, posso solo aprirle), e invece rimane sempre in quella condizione

Sicuramente la mia ignoranza da bravo "dilettante-alla-prime-armi" non mi aiuta, ma non sono ancora riuscito a trovare nulla online che mi aiuti nel trovare la soluzkione del problema.

**Comment by dante70** [comment]
ciao,
quando scrivi che i tuoi scuri richiedono una pressione istantanea per poi chiudersi /aprirsi da soli significa che hanno un loro circuito che li gestisce?
Se la risposta  è si secondo me il sistema non può darti  la corretta "visualizzazione " in quanto tu no riesci a fare la calibrazione perchè lo shelly 2.5 non sente assorbimento di corrente a sufficienza (perchè non pilota i motori direttamente).
Ho appena finito di montare 9 shelly per le mie tapparelle e son abbastanza fresco  di configurazione ......se serve facciamo delle prove sui miei apparecchi

**Reply by Jocman** [reply]
Dopo aver smanettato un pò tutto il pomeriggio/sera, penso di aver completamente sbagliato ad impostare tutta la problematica.
E stamattina, dopo aver letto la documentazione degli scuri, penso di essermi convinto dell'errore.
Rispondendo a Dante, SI, ogni finestra ha una propria centralina che gestisce gli scuri.
La mia idea era semplicemente quella di collegare in parallelo ai pulsanti esistenti un sistema che simulasse la pressione e aprisse/chiudesse gli scuri.
Per fare ciò mi sarebbero bastati 2 relays a contatto pulito, e pensavo che gli shelly 2.5 facessero al caso mio.
Ma alla fine ritengo non sia possibile, in quanto gli shelly, guardando uno schema tipico di collegamento alle persiane, a seconda del pulsante premuto da corrente direttamente al motore apre/chiude (come giustamente Dante fa notare)
Quindi, credo, dovrei eliminare la scheda elettronica degli scuri e collegare gli shelly direttamente ai motori (sono 2), ma sinceramente la cosa non mi alletta.
La soluzione potrebbe essere far funzionare gli shelly non in "Shutter mode", ma in "Switch mode", peccato che non supporti una modalità "momentary switch", e la documentazione della centralina avverte espressamente di non esercitare una pressione prolungata sul pulsante (nè tantomento di utilizzare dei normali toggle invece di momentary).
A meno che non mi possiate indicare una soluzione, l'unica che sto accarezzando è quella di autocostruirmi un sistema utilizzabile nel mio caso, e pensavo a un semplice modulo 2 relays pilotato da ESP32.

**Reply by bf69** [reply]
ciao, a me sembra che si possa impostare gli ingressi dello shelly come momentaneo in modalità relays...

**Reply by Jocman** [reply]
Uhmmm girando per la configurazione non mi pare di aver trovato il modo, ma ci darò di nuovo un'occhiata

**Reply by Jocman** [reply]
@bf69 
Mi è arrivata la mail di notifica della tua risposta con l'immagine allegata, ma qui nel forum non c'è.... boh...
In ogni caso quando torno a casa proverò con il tuo suggerimento.

**Reply by Jocman** [reply]
Allora, ho provato a fare come suggerito, ma:
- Nel menu dei miei Shelly 2.5 non esiste quel tipo di configurazione
- tra l'altro, il menu è in inglese (non che sia un problema, affatto, ma vederlo in italiano fa un altro effetto)
Mi è venuto il sospetto che ci sia una differenza di firmware, e infatti mi da come disponibile un nuovo firmware aggiornato.
Dopo aver fatto l'aggiornamento, mi è comparso un menu molto più ricco di voci, tra le quali anche quella suggerita.
Peccato che fosse scomparso un canale (quindi visualizzavo un solo relay)
Ho pensato di fare un reboot dello shelly, e sono ricomparsi entrambe i relays; peccato che il menu sia tornato quello di prima, cioè senza le voci indicate.
Queste sono le uniche voci disponibili nel menu:

![screenshot 2](https://user-images.githubusercontent.com/52747587/115589502-3588ff00-a2d0-11eb-8285-49c66d10cb0e.jpg)

**Reply by dante70** [reply]
ciao, forse mi sfuggirà qualcosa ma io non ho capito il problema.....
se non vuoi  collegare direttamente gli shelly ai motori degli scuri non serve che si stia a pensare le varie configurazioni dei pulsanti in  quanto   non avendo i rele' a contatto pulito non puoi metterli in parallelo ai pulsanti che hai ora.
questo vale se l'attuale gestione degli scuri è fatta con un circuito elettronico separato dalla rete con ingressi pilotati dai pulsanti attuali
Se invece puoi collegare la rete 220v io penso che con la modalità che hai indicato al primo post e con tempo di 1'' il problema non sussista.
Ho appena provato uno shelly con tempi di 1 sec e  sull'interfaccia ho tutte e 3 le icone(freccia alta,bassa e quadratino di stop) e posso aprire e chiudere regolarmente e sento  i relè sganciarsi dopo il tempo impostato........

**Reply by Jocman** [reply]
A questo punto mi sa che, nella scelta degli shelly sono stato messo nella direzione sbagliata dalla loro descrizione sul sito, che benchè non riporti esplicitamente il tipo di contatto, ad un certo punto dice che si possono utilizzare al posto di due shelly 1 (che specifica essere a contatto non pulito).
Comunque, dato per scontato il mio errore di base, poichè come detto non voglio smantellare tutta l'elettronica di controllo già presente, mi orienterò ad altra soluzione e gli shelly li userò (spero) per altro.
Quanto al problema delle icone, invece ti confermo che ne vedo attive solo due, come nella foto che ho postato all'inizio.
A questo punto mi è rimasta solo la curiosità di capirne il motivo.
